src/network/networkd-netdev-tunnel.c

```diff
@@ -56,7 +56,7 @@ static int netdev_ipip_fill_message_create(NetDev *netdev, Link *link, sd_netlin
         assert(link);
         assert(m);
         assert(t);
-        assert(t->family == AF_INET);
+        assert(t->family == AF_INET || t->family != -1);
```

Comment:
This doesn't look right. I think you meant IN_SET(t->family, AF_INET, 0);

It would be also nice to replace the 0 by AF_UNSPEC, here and elsewhere.